tests/cupy_tests/logic_tests/test_comparison.py

```diff
@@ -137,6 +137,56 @@ def test_array_equal_broadcast_not_allowed(self, xp):
         return xp.array_equal(a, b)
 
 
+class TestArrayEquiv(unittest.TestCase):
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_equal_not_equal(self, xp, dtype):
+        a = xp.array([0, 4, 1, 1], dtype=dtype)
+        b = xp.array([0, 4, 2, 3], dtype=dtype)
+        return xp.array_equiv(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_equal_is_equal(self, xp, dtype):
+        a = xp.array([0, 4, 1, 1], dtype=dtype)
+        b = xp.array([0, 4, 1, 1], dtype=dtype)
+        return xp.array_equiv(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_diff_length(self, xp, dtype):
+        a = xp.array([0, 4, 0, 5], dtype=dtype)
+        b = xp.array([0, 4, 0], dtype=dtype)
+        return xp.array_equiv(a, b)
+
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_diff_dtypes_not_equal(self, xp):
+        a = xp.array([0.9e-5, 1.1e-5, 100.5, 10.5])
+        b = xp.array([0, 0, 1000, 1000])
+        return xp.array_equiv(a, b)
+
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_diff_dtypes_is_equal(self, xp):
+        a = xp.array([0.0, 1.0, 100.0, 10.0])
+        b = xp.array([0, 1, 100, 10])
+        return xp.array_equiv(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_broadcast(self, xp, dtype):
+        a = xp.array([0, 4], dtype)
+        b = xp.array([[0, 4], [0, 4]], dtype)
+        return xp.array_equiv(a, b)
+
+    @testing.for_all_dtypes()
+    @testing.numpy_cupy_equal()
+    def test_array_equiv_broadcast_not_equal(self, xp, dtype):
+        a = xp.array([0, 4], dtype=dtype)
+        b = xp.array([[0, 4], [0, 5]], dtype=dtype)
+        return xp.array_equiv(a, b)
+
```

Comment:
Please add a test case to cover that given two arrays are not broadcastable.

Reply:
Hi, @kmaehashi,
I think the above tests depict the same. Array `xp.array([0, 4], dtype=dtype)` and the array `b = xp.array([[0, 4], [0, 5]], dtype=dtype)` are not broadcastable.
Would you please tell me if I'm wrong?

But yes, I agree the name of the definition can be made more explicit. What do you think?

(I also added one more test for non-broadcastable arrays with different dimensions.)

Reply:
`xp.array([0, 4])` and `xp.array([[0, 4], [0, 5]])` are broadcastable 🙂

```py
>>> xp.array([0, 4]) + xp.array([[0, 4], [0, 5]])
array([[0, 8],
       [0, 9]])
```

Could you fix the test case name?

Reply:
Ah, of course. It's broadcastable. I had a couple of different things in my mind at that time.